contrib/dockerd-rootless.sh

```diff
@@ -74,6 +74,10 @@ fi
 if [ -z $_DOCKERD_ROOTLESS_CHILD ]; then
 	_DOCKERD_ROOTLESS_CHILD=1
 	export _DOCKERD_ROOTLESS_CHILD
+	if [ "$(id -u)" = "0" ]; then
+		echo "Must not run as root"
```

Comment:
Thought I left a comment, but I guess I forgot.

Does this only apply to this part of the code, or to the script as a whole? If it's for the whole script, perhaps something like;

```suggestion
		echo "This script must be executed as a non-privileged user"
```

(and in that case, I think it would make sense to put it at the top of the script, together with the other checks at the start)

Reply:
Updated the error message.

> put it at the top of the script, together with the other checks at the start

No. The script "reexecs" itself inside a user namespace, so the UID flips to be zero.